src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -126,8 +126,15 @@ class IndexLogEntryTest extends SparkFunSuite with SQLHelper with BeforeAndAfter
         |                  "properties" : { }
         |                }
         |              },
-        |              "deletedFiles" : ["file:/rootpath/f1"],
-        |              "appendedFiles" : ["file:/rootpath/f3"]
+        |              "deletedFiles" : [{
+        |                    "name" : "file:/rootpath/f1",
```

Comment:
Now that I look that this, should we store these as `Directory` to be consistent? The thing that bugs me is that `FileInfo`'s behavior is inconsistent - one is storing full path, and another just name. Also, this can save the metadata file size. Thoughts?

(I know we are using it interchangeably in the code - e.g., `Content.fileInfos`, which can be addressed separately).

Reply:
I also thought about using `Content` as utility functions are coupled with it, not with `Directory`. (and inconsistency of FileInfo also bothers me). However, the format below didn't seem good to me, a little. Do you think this is more preferable?

```
case class Hdfs(properties: Hdfs.Properties) {
  val kind = "HDFS"
}
object Hdfs {
  /**
   * Hdfs file properties.
   * @param content Content object representing Hdfs file based data source.
   * @param appendedFiles Appended files since the last time derived dataset was updated.
   * @param deletedFiles Deleted files since the last time derived dataset was updated.
   */
  case class Properties(
      content: Content,
      appendedFiles: Content,
      deletedFiles: Content)
}
```

or with #198
```
case class Update {
  appendedFiles: Content,
  deletedFiles: Content,
  latestSignature: Signature
}
```